common/djangoapps/lang_pref/middleware.py

```diff
@@ -26,3 +28,12 @@ def process_request(self, request):
             # Set it to the LANGUAGE_SESSION_KEY (Django-specific session setting governing language pref)
             if user_pref:
                 request.session[LANGUAGE_SESSION_KEY] = user_pref
+            else:
+                # Set browser language if it is supported by system.
+                preferred_language = request.META.get('HTTP_ACCEPT_LANGUAGE', '')
+                lang_headers = [seq[0] for seq in parse_accept_lang_header(preferred_language)]
+                for browser_lang in lang_headers:
+                    if browser_lang in [seq[0] for seq in released_languages()]:
```

Comment:
It might be better to set a "released_languages" parameter outside of the loop for performance reasons.